### PR TITLE
Update authoring for sqlitepcl 2.0.4

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Note: we may need to update https://github.com/dotnet/roslyn/blob/master/src/Setup/Roslyn.ThirdPartyNotices/ThirdPartyNotices.rtf

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -225,6 +225,7 @@
     <SystemValueTupleVersion>4.5.0</SystemValueTupleVersion>
     <SystemThreadingTasksExtensionsVersion>4.5.4</SystemThreadingTasksExtensionsVersion>
     <SQLitePCLRawbundle_greenVersion>2.0.4</SQLitePCLRawbundle_greenVersion>
+    <SQLitePCLRawproviderdynamic_cdeclVersion>2.0.4</SQLitePCLRawproviderdynamic_cdeclVersion>
```
Comment (genlu) [inline eng/Versions.props:228]: Is this supposed to be referenced somewhere?

Comment (CyrusNajmabadi) [reply eng/Versions.props:228]: @jasonmalinowski ?

Comment (jasonmalinowski) [reply eng/Versions.props:228]: Yeah this isn't the right change to make, you need to edit the NuGetPackageToIncludeInVsix stuff I linked to in our private conversation.

**Comment by davidwengier** [comment]
Was there a build error that brought up the need this, or runtime?

I'm trying to sort out updated SQLite dependencies in VS Mac so any extra context would be appreciated

**Review by sharwell — APPROVED** [review]

**Comment by sharwell** [comment]
I extracted the VSIX and verified that both:

* The new assembly SQLitePCLRaw.provider.dynamic_cdecl.dll is being included
* The new assembly is registered in Roslyn.VisualStudio.Setup.pkgdef

**Review by sharwell — APPROVED** [review]

**Review by AArnott — APPROVED** [review]

File: src/Workspaces/Core/Portable/Storage/SQLite/AbstractSQLitePersistentStorageService.cs [hunk 2]
```diff
@@ -16,35 +16,12 @@ namespace Microsoft.CodeAnalysis.SQLite
...
-                if (myFolder == null)
-                    return false;
-
-                var is64 = IntPtr.Size == 8;
-                var subfolder = is64 ? "x64" : "x86";
-
-                LoadLibrary(Path.Combine(myFolder, subfolder, "e_sqlite3.dll"));
-            }
-
             try
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Storage/SQLite/AbstractSQLitePersistentStorageService.cs:25]: Does the catch block for this try need to be updated as well to not be filtering exceptions? It's unclear to me why that ever was there, but it seems like that too should have caught these exceptions all along.

**Review by jasonmalinowski — APPROVED** [review]
:shipit:, but as a follow-up lets make sure we get an integration test covering this.

**Pull request merged**